resources/rebuild.sh

```diff
@@ -134,80 +134,74 @@ EOF
-    git clean -fdx
-    git checkout $LATEST_TAG
+function clone_amazon_linux_repo {
+  [ -d linux ] || git clone https://github.com/amazonlinux/linux linux
 }
 
-
-# Download the latest kernel source for the given kernel version
-function get_linux_tarball {
-    local KERNEL_VERSION=$1
-    echo "Downloading the latest patch version for v$KERNEL_VERSION..."
-    local major_version="${KERNEL_VERSION%%.*}"
-    local url_base="https://cdn.kernel.org/pub/linux/kernel"
-    local LATEST_VERSION=$(
-        curl -fsSL $url_base/v$major_version.x/ \
-        | grep -o "linux-$KERNEL_VERSION\.[0-9]*\.tar.xz" \
-        | sort -rV \
-        | head -n 1 || true)
-    # Fetch tarball and sha256 checksum.
-    curl -fsSLO "$url_base/v$major_version.x/sha256sums.asc"
-    curl -fsSLO "$url_base/v$major_version.x/$LATEST_VERSION"
-    # Verify checksum.
-    grep "${LATEST_VERSION}" sha256sums.asc | sha256sum -c -
-    echo "Extracting the kernel source..."
-    tar -xaf $LATEST_VERSION
-    local DIR=$(basename $LATEST_VERSION .tar.xz)
-    ln -svfT $DIR linux
+# prints the git tag corresponding to the newest and best matching the provided kernel version $1
+# this means that if a microvm kernel exists, the tag returned will be of the form
+#
+#    microvm-kernel-$1.<patch number>.amzn2[023]
+#
+# otherwise choose the newest tag matching
+#
+#    kernel-$1.<patch number>.amzn2[023]
+function get_tag {
+  local KERNEL_VERSION=$1
+
+  # list all tags from newest to oldest
+  (git --no-pager tag -l --sort=-creatordate | grep "microvm-kernel-$1\..*\.amzn2" \
+    || git --no-pager tag -l --sort=-creatordate | grep "kernel-$1\..*\.amzn2") | head -n1
 }
 
-function build_linux {
-    local KERNEL_CFG=$1
-    # Extract the kernel version from the config file provided as parameter.
-    local KERNEL_VERSION=$(grep -Po "^# Linux\/\w+ \K(\d+\.\d+)" "$KERNEL_CFG")
-
-    get_linux_tarball $KERNEL_VERSION
-    pushd linux
-
-    arch=$(uname -m)
-    if [ "$arch" = "x86_64" ]; then
-        format="elf"
-        target="vmlinux"
-        binary_path="$target"
-    elif [ "$arch" = "aarch64" ]; then
-        format="pe"
-        target="Image"
-        binary_path="arch/arm64/boot/$target"
-
-        # Patch 6.1 kernels on ARM with 6.10 patches for supporting VMGenID
-        # via DeviceTree bindings.
-        # TODO: drop this (and remove the patches from the repo) when we switch
-        # to building kernels from AL tree.
-        if [[ $KERNEL_VERSION == "6.1" ]]; then
-            for i in ../patches/vmgenid_dt/* ; do
-                patch -p1 < $i
-            done
-        fi
-    else
-        echo "FATAL: Unsupported architecture!"
-        exit 1
+function build_al_kernel {
+  local KERNEL_CFG=$1
+  # Extract the kernel version from the config file provided as parameter.
+  local KERNEL_VERSION=$(echo $KERNEL_CFG | grep -Po "microvm-kernel-ci-$ARCH-\K(\d+\.\d+)")
+
+  pushd linux
+  make clean
+  make distclean
```

Comment:
why do we need to call both? isn't `make distclean` enough?
as per [Makefile](https://github.com/torvalds/linux/blob/master/Makefile#L1475), `distclean` depends on `mrproper` and `mrproper` depends on `clean`.

Reply:
Ah, cool. I think I just saw some instructions at some point in my life that had both, and have been doing both ever since haha